mypy/semanal.py

```diff
@@ -1216,7 +1216,8 @@ def analyze_function_body(self, defn: FuncItem) -> None:
 
     def check_classvar_in_signature(self, typ: ProperType) -> None:
         if isinstance(typ, Overloaded):
-            for t in typ.items:  # type: ProperType
+            t: ProperType
```

Comment:
I think we can remove this one, `_items` is `list[CallableType]`, which is a subtype of `ProperType`: https://github.com/python/mypy/blob/551f8f4064c2158d47917b418726a01a2a797a7d/mypy/types.py#L1979

Reply:
The declaration acts as an upcast here. If you remove it, the selfcheck fails with this message:

```
mypy\semanal.py:1224: error: Incompatible types in assignment (expression
has type "ProperType", variable has type "CallableType")  [assignment]
            for t in get_proper_types(typ.arg_types) + [get_proper_typ...
```

It would be worthwhile to move the declaration, though -- it shouldn't really be in this `if` block 👍